subsys/bluetooth/host/settings.c

```diff
@@ -281,7 +281,8 @@ static int commit_settings(void)
 	return 0;
 }
 
-SETTINGS_STATIC_HANDLER_DEFINE(bt, "bt", NULL, set_setting, commit_settings, NULL);
+/* Add `_0` prefix to ensure BT always commit firstly. */
+SETTINGS_STATIC_HANDLER_DEFINE(_0_bt, "bt", NULL, set_setting, commit_settings, NULL);
```

Comment:
That's not really a good solution IMO

If something is `00_` then it still comes before `0_bt`. 

Unfortunately there doesn't seem to be any maintainer for the settings subsystem, but if we want to add priorities or specific order of settings, then I think we need to consider better alternatives. 

At the top of my head, suggestions could be

1. Add a priority in the API (either the BT_SETTINGS or SETTINGS API)
2. Have a single settings handler in BT that can load the others in the right order. 

This solution here seems to use an undocumented approach that is easy to get wrong and is based on the assumption that the names will always control the order

Reply:
Done.

Reply:
``` C
 *(SORT_BY_NAME(SORT_BY_ALIGNMENT(._settings_handler_static.static.*)))
 ._settings_handler_static.static.settings_handler_0_bt_
                0x0000000000034644       0x14 zephyr/subsys/bluetooth/host/libsubsys__bluetooth__host.a(settings.c.obj)
                0x0000000000034644                settings_handler_0_bt
 ._settings_handler_static.static.settings_handler_0_bt_ccc_
                0x0000000000034658       0x14 zephyr/subsys/bluetooth/host/libsubsys__bluetooth__host.a(gatt.c.obj)
                0x0000000000034658                settings_handler_0_bt_ccc
 ._settings_handler_static.static.settings_handler_0_bt_cf_
                0x000000000003466c       0x14 zephyr/subsys/bluetooth/host/libsubsys__bluetooth__host.a(gatt.c.obj)
                0x000000000003466c                settings_handler_0_bt_cf
 ._settings_handler_static.static.settings_handler_0_bt_hash_
                0x0000000000034680       0x14 zephyr/subsys/bluetooth/host/libsubsys__bluetooth__host.a(gatt.c.obj)
                0x0000000000034680                settings_handler_0_bt_hash
 ._settings_handler_static.static.settings_handler_0_bt_sc_
                0x0000000000034694       0x14 zephyr/subsys/bluetooth/host/libsubsys__bluetooth__host.a(gatt.c.obj)
                0x0000000000034694                settings_handler_0_bt_sc
 ._settings_handler_static.static.settings_handler_1_bt_mesh_
                0x00000000000346a8       0x14 zephyr/subsys/bluetooth/mesh/libsubsys__bluetooth__mesh.a(settings.c.obj)
                0x00000000000346a8                settings_handler_1_bt_mesh
 ._settings_handler_static.static.settings_handler_1_bt_mesh_app_
                0x00000000000346bc       0x14 zephyr/subsys/bluetooth/mesh/libsubsys__bluetooth__mesh.a(app_keys.c.obj)
                0x00000000000346bc                settings_handler_1_bt_mesh_app
 ._settings_handler_static.static.settings_handler_1_bt_mesh_cfg_
                0x00000000000346d0       0x14 zephyr/subsys/bluetooth/mesh/libsubsys__bluetooth__mesh.a(cfg.c.obj)
                0x00000000000346d0                settings_handler_1_bt_mesh_cfg
 ._settings_handler_static.static.settings_handler_1_bt_mesh_comp_
                0x00000000000346e4       0x14 zephyr/subsys/bluetooth/mesh/libsubsys__bluetooth__mesh.a(access.c.obj)
                0x00000000000346e4                settings_handler_1_bt_mesh_comp
 ._settings_handler_static.static.settings_handler_1_bt_mesh_iv_
                0x00000000000346f8       0x14 zephyr/subsys/bluetooth/mesh/libsubsys__bluetooth__mesh.a(net.c.obj)
                0x00000000000346f8                settings_handler_1_bt_mesh_iv
 ._settings_handler_static.static.settings_handler_1_bt_mesh_net_
                0x000000000003470c       0x14 zephyr/subsys/bluetooth/mesh/libsubsys__bluetooth__mesh.a(net.c.obj)
                0x000000000003470c                settings_handler_1_bt_mesh_net
```